protoc-gen-eventuals/templates/grpc-method-eventuals.cc.j2

```diff
@@ -0,0 +1,128 @@
+{%- set service = services[service_number] %}
+{%- set method = service.methods[method_number] %}
```

Comment:
Are you using `service_number` or `method_number` anywhere else?

If not, you can change skip all the `enumerate(...)` you do in python and just do:

```python
for service in template_data['services']:
    for method in service['methods']:
    	# ...
	service_data = {
	    'service': service,
	    'method': method,
	}
	# ...
	content = template.render(**template_data, **service_data)
```

and this would give you the variable `service` and `method` in your jinja.

Reply:
I like this a lot!